Mage.Sets/src/mage/cards/i/IrreverentGremlin.java

```diff
@@ -0,0 +1,60 @@
+package mage.cards.i;
+
+import java.util.UUID;
+
+import mage.MageInt;
+import mage.abilities.common.EntersBattlefieldControlledTriggeredAbility;
+import mage.abilities.costs.common.DiscardCardCost;
+import mage.abilities.effects.common.DoIfCostPaid;
+import mage.abilities.effects.common.DrawCardSourceControllerEffect;
+import mage.abilities.keyword.MenaceAbility;
+import mage.cards.CardImpl;
+import mage.cards.CardSetInfo;
+import mage.constants.CardType;
+import mage.constants.ComparisonType;
+import mage.constants.SubType;
+import mage.filter.common.FilterCreaturePermanent;
+import mage.filter.predicate.mageobject.AnotherPredicate;
+import mage.filter.predicate.mageobject.PowerPredicate;
+
+/**
+ * @author paasar
+ */
+public final class IrreverentGremlin extends CardImpl {
+    private static final FilterCreaturePermanent creatureWithPower2OrLessFilter = new FilterCreaturePermanent("creature with power 2 or less");
+
+    static {
+        creatureWithPower2OrLessFilter.add(AnotherPredicate.instance);
+        creatureWithPower2OrLessFilter.add(new PowerPredicate(ComparisonType.FEWER_THAN, 3));
+
+    }
+
+    public IrreverentGremlin(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.CREATURE}, "{1}{R}");
+
+        this.subtype.add(SubType.GREMLIN);
+        this.power = new MageInt(2);
+        this.toughness = new MageInt(2);
+
+        // Menace (This creature can't be blocked except by two or more creatures.)
+        this.addAbility(new MenaceAbility());
+
+        // Whenever another creature you control with power 2 or less enters, you may discard a card. If you do, draw a card. Do this only once each turn.
+        this.addAbility(
+                new EntersBattlefieldControlledTriggeredAbility(
+                        new DoIfCostPaid(
+                                new DrawCardSourceControllerEffect(1),
+                                new DiscardCardCost()),
+                        creatureWithPower2OrLessFilter)
+                        .setTriggersLimitEachTurn(1));
```

Comment:
This isn't correct. Card text is "Do only once each turn" not "triggers only once each turn".

As an additional complication, to get this right, the trigger has to be optional but the DoIfCostPaid not optional

Reply:
@xenohedron Can you explain the diff? It’s a standard rules notation for trigger/choice limitation and used [all around](https://scryfall.com/search?q=o%3A%22Do+this+only+once+each+turn%22&unique=cards&as=grid&order=name).

Example: [[Corruption of Towashi]]
> Whenever a permanent you control transforms or a permanent you control enters transformed, you may draw a card. Do this only once each turn.

It has oracle text about trigger limitation (limit will be applied after first “yes” usage):
> Once you’ve chosen to draw a card because of the last ability (which is likely as soon as possible, but there may be situations in which you don’t want to draw immediately), that ability will stop triggering for the duration of that turn.
(2023-04-14)

And it works as optional inside:
![IMG_0925](https://github.com/user-attachments/assets/44a76e3a-00d7-4c33-8afa-f01a603db874)

Reply:
`setDoOnlyOnceEachTurn(true)` is correct. The code I reviewed uses `setTriggersLimitEachTurn(1)` which is not the same.

Reply:
"Do only once" means you can decline and it will still trigger later. "Triggers only once" would mean that it only triggers once regardless of whether you choose to take action that time.